test/composertest.py

```diff
@@ -81,9 +81,21 @@ def runCliTest(self, script):
         self.assertEqual(r.returncode, 0)
 
 
+def print_tests(tests):
+    for test in tests:
+        if isinstance(test, unittest.TestSuite):
+            print_tests(test)
+        elif isinstance(test, unittest.loader._FailedTest):
+            name = test.id().replace("unittest.loader._FailedTest.", "")
+            print(f"Error: '{name}' does not match a test", file=sys.stderr)
+        else:
+            print(test.id().replace("__main__.", ""))
+
+
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument("tests", nargs="*", help="List of tests modules, classes, and methods")
+    parser.add_argument("-n", "--dry-run", action="store_true", help="Print the list of tests that would be executed")
```

Comment:
Complete bikeshedding of course, but what about `-l`/`--list-tests`? "run" is a bit confusing as this doesn't run anything. glib's tests use `-l` (it's a standard option)

Reply:
I prefer that, too. Changed.